### PR TITLE
fix(bundle/definition): add custom validator for contentEncoding

### PR DESCRIPTION
**Pull request by vdice**
This represents a fix to ensure definitions with a non-empty `contentEncoding` property pass validation by adding a custom validator for continued use of the jsonschema library cnab-go currently uses.  It's specific to `contentEncoding` but illustrates a potential path forward for further "custom" validators (either for properties in the default JSON Schema but *not* stock in the jsonschema library or for truly custom properties/validators down the line.)

See https://github.com/deislabs/cnab-go/issues/117 for background on general issue and https://github.com/deislabs/porter/pull/579 for feature that uses the changes here to pass validation.

### CODE REVIEW

File: bundle/definition/validators.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// Validate implements the Validator interface for ContentEncoding
+// Currently, this is a no-op and is only used to register with the jsonschema library
+// that 'contentEncoding' is a valid property (as of writing, it isn't one of the defaults)
+func (c ContentEncoding) Validate(propPath string, data interface{}, errs *[]jsonschema.ValError) {}
+
+// NewRootSchema returns a jsonschema.RootSchema with any needed custom
+// jsonschema.Validators pre-registered
+func NewRootSchema() *jsonschema.RootSchema {
+	// Register custom validators here
+	// Note: as of writing, jsonschema doesn't have a stock validator for intances of type `contentEncoding`
```
Comment (glyn) [inline bundle/definition/validators.go]:
```suggestion
	// Note: as of writing, jsonschema doesn't have a stock validator for instances of type `contentEncoding`
```

File: bundle/definition/validation_test.go [hunk 2]
```diff
@@ -42,6 +42,39 @@ func TestObjectValidationValid(t *testing.T) {
...
+	assert.Equal(t, 1, len(props), "should have had a single property")
+	propSchema, ok := props["file"]
+	assert.True(t, ok, "should have found file property")
+	assert.Equal(t, "string", propSchema.Type, "file type should have been a string")
+	assert.Equal(t, "base64", propSchema.ContentEncoding, "file contentEncoding should have been base64")
+
+	val := struct {
+		File string `json:"file"`
+	}{
+		File: "SGVsbG8gV29ybGQhCg==",
```
Comment (glyn) [inline bundle/definition/validation_test.go]: If I change this to an invalid base64 value such as "SGVsbG8gV29ybGQhCg===", the test still passes. Shouldn't it fail?

File: bundle/definition/validators.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package definition
+
+import (
+	"github.com/qri-io/jsonschema"
+)
+
+// ContentEncoding represents a "custom" Schema property
+type ContentEncoding string
```
Comment (silvin-lubecki) [inline bundle/definition/validators.go]: As `NewContentEncoding` is returning a `jsonschema.Validator`, do we really need to expose that type?

File: bundle/definition/validation_test.go (hunk 2 above)
Comment (vdice) [reply bundle/definition/validation_test.go]: Good call; indeed, as mentioned in the code comments, the current validator is a no-op (simply _allows_ for a `contentEncoding` property), but we can and perhaps should actually perform some basic validations specific to this property.  Preference on doing so in this PR or as a follow-up?

Comment (glyn) [reply bundle/definition/validation_test.go]: Actually, I'd prefer it in this PR to check feasibility _before_ merging this PR. Hope it's not too much work - maybe a separate commit?

Comment (glyn) [reply bundle/definition/validation_test.go]: I think the values we need to support are defined by https://json-schema.org/latest/json-schema-validation.html#rfc.section.8.3.

Comment (vdice) [reply bundle/definition/validation_test.go]: Ok, validation added for `base64`.  Wanted to keep scope limited so didn't venture off into adding other content encoding validations quite yet.  What do you think?

Comment (glyn) [reply bundle/definition/validation_test.go]: Fair enough. Why not raise issue(s) for the other encodings?

Comment (vdice) [reply bundle/definition/validation_test.go]: Deal! https://github.com/deislabs/cnab-go/issues/132

File: bundle/definition/validators.go (hunk 3 above)
Comment (silvin-lubecki) [reply bundle/definition/validators.go]: I don't have strong opinion on it, I just have the habit of reducing the exposed parts to their minimum 👍

**Pull request merged**